src/app/main/component/user/components/profile/profile-service/profile.service.ts

```diff
@@ -46,6 +47,32 @@ export class ProfileService {
     return this.http.get<FactOfTheDay>(`${mainLink}fact-of-the-day/random?lang=${currentLang}`);
   }
 
+  getHabitFactFromLocalStorage(): FactOfTheDay | null {
+    const savedHabitFact = localStorage.getItem('habitFactOfTheDay');
+    const lastHabitFetchTime = localStorage.getItem('lastHabitFactFetchTime');
+    const currentTime = Date.now();
+
+    if (savedHabitFact && lastHabitFetchTime && currentTime - Number(lastHabitFetchTime) < this.oneDayInMillis) {
```

Comment:
You can simplify the condition and break it down into smaller, easier to read parts